monorepo/monorepo-ws/apps/federation-gateway/src/services/server-manager.ts

```diff
@@ -1,52 +1,48 @@
 import { ApolloGateway } from "@apollo/gateway";
 import { ApolloServer } from "apollo-server";
-import { checkActualSuperGraph } from ".";
 import { interval, port } from "../configuration/config";
-import { SupergraphStorageManager } from "./interfaces";
 import { tryCreatePartOfSupergraph } from "./supergraph-services";
+import { SupergraphManager } from "@monorepo-ws/supergraph-manager"
+import { logger } from "libs/server/src/logger";
 
 
 export class ServerManager {
-    superGraphManager: SupergraphStorageManager
-    superGraphSchema: Buffer
     gateway: ApolloGateway
     server: ApolloServer
 
-    constructor(superGraphManager: SupergraphStorageManager){
-        this.superGraphManager = superGraphManager
-    }
-
-    private async checkSupergraph(){
-        const newFile = await this.superGraphManager.getSupergraph()
-        if(!checkActualSuperGraph(this.superGraphSchema, newFile)){
-            await this.restartServer(newFile)
-        }
+    constructor(private readonly supergraphManager: SupergraphManager){
     }
 
     async runApp() {
         await this.runServer()
-        setInterval(() => this.checkSupergraph, interval)
+        setInterval(this.runServer.bind(this), interval)
     }
 
     private async runServer() {
+        if(!this.server){
+            try {
+                const superGraphSchema: Buffer = await this.supergraphManager.getSupergraph()
+                return await this.startServer(superGraphSchema.toString())
+            } catch (error) {
+                const superGraphSchema: Buffer =  await tryCreatePartOfSupergraph() 
+                return await this.startServer(superGraphSchema.toString())
+            }
+        }
         try {
-            this.superGraphSchema = await this.superGraphManager.getSupergraph()
-          } catch (error) {
-            this.superGraphSchema =  await tryCreatePartOfSupergraph() 
-          }
-          this.gateway = new ApolloGateway({ supergraphSdl: this.superGraphSchema.toString() });
-          
-          this.server = new ApolloServer({ gateway: this.gateway });
-          this.server.listen({ port: port })
+            const superGraphSchema = await this.supergraphManager.getSupergraph()
+            if(!this.supergraphManager.checkSupergraph(superGraphSchema)){
+                this.supergraphManager.setActualSupergraph(superGraphSchema)
+                await this.server.stop()                
+                this.startServer(superGraphSchema.toString())
+            }
+        } catch (error) {  
+            logger.info("storage is probably unavailable") 
```

Comment:
❗️tohle urcite nikdy nedelej, predstav si ze ti ta appka fakt bezi a najednou tam neco nejde a ty jedine co se dozvis v logach je "storage is probably ..." aj este probably, ale tech duvodu co se stalo muze bejt cela rada 

zamyslel bych se i nad tim jestli fakt nelogovat error

co s tim:
- vlastni error kterej ma i vlastni message a prebira prevError
- proste zalogovat logger.errro